docs/architecture/tablets.rst

```diff
@@ -0,0 +1,99 @@
+cluster quickly and seamlessly.
+
+Data Distribution
+-------------------
+
+ScyllaDB distributes data by splitting tables into tablets. Each tablet has 
+its replicas on different nodes, depending on the RF (replication factor). Each
+partition of a table is mapped to a single tablet in a deterministic way. When you
+query or update the data, ScyllaDB can quickly identify the tablet that stores
+the relevant partition. 
+
+The following example shows a 3-node cluster with a replication factor (RF) of
+3. The data is stored in a table (Table 1) with two rows. Both rows are mapped
+to one tablet (T1) with replicas on all three nodes.
+
+.. image:: images/tablets-cluster.png
+
+.. TODO - Add a section about tablet splitting when there are more triggers,
+   like throughput. In 6.0, tablets only split when reaching a threshold size
+   (measured in the number of rows).
+
+Load Balancing
+==================
+
+ScyllaDB autonomously moves tablets to balance the load based on table usage. 
+This process is managed by a load balancer mechanism and happens independently
+of the administrator or node operations such as bootstrap or decommission.
+The tablet load balancer decides where to migrate the tablets, either within
+the same node to balance the shards or across the nodes to balance the global
+load in the cluster. 
+
+The load-balancing process takes place in the background and is performed
+without any service interruption.
+
+Scaling Out
+=============
+
+A tablet can be dynamically migrated to either an existing node or a new node –
+on bootstrap. Tablets make scaling out fast and efficient, as the new node
+starts taking part of the load even before the bootstrap process completes.
+Also, tablets paired with consistent topology updates with Raft allow you to
+add multiple nodes simultaneously. Existing nodes stream data to the new ones,
+and the system load eventually converges to an even distribution as the process
+completes.
+
+The following diagrams show migrating tablets from heavily loaded nodes A and B
+to a new node.
+
+.. image:: images/tablets-load-balancing.png
+
+.. _tablets-enable-tablets: 
+
+Enabling Tablets
+-------------------
+
+Tablets are enabled or disabled on the keyspace level. When you create a new
+keyspace, tablets are enabled by default. You can create a keyspace with tablets
+disabled with the ``tablets_enabled:false`` option:
+
+.. code:: cql
+
+    CREATE KEYSPACE my_keyspace
+    WITH replication = {
+        'class': 'NetworkTopologyStrategy',
+        'replication_factor': 3,
+    } AND tablets_enabled = false;
+
+.. warning::
+
+    You cannot ALTER a keyspace to enable or disable tablets.
+    The only way to update the tablet support for a keyspace is to DROP it
+    (losing the schema and data) and then recreate it from a backup.
+
+Unsupported Features
+----------------------
+
+The following ScyllaDB features are not supported if a keyspace has tablets
+enabled:
+
+* Counters
+* LWT
+* CDC
```

Comment:
Also, resharding with tablets has caveats in 6.0.
1. ScyllaDB does not support scaling down the number of shards when tablets are allocated to shards that do not exist after backup.  The plan is to provide a way to "decommission" those shards while the node is still up so to migrate tablets away from them to a smaller set of shards and only then to restart the node with fewer shards.
2. For scale-up, unlike vnodes that reshards the data synchronously when the node restarts, tablets-enabled tables will currently stay put on restart and will be subject to background load-balancing towards the additional shards after the node restarts and starts serving CQL.